libr/core/cmd.c

```diff
@@ -6836,10 +6836,25 @@ R_API int r_core_cmd_lines(RCore *core, const char *lines) {
 	if (!odata) {
 		return false;
 	}
+	int line_count = 0;
+	{
+		int i = 0;
+		while (lines[i]) {
+			if (lines[i] == '\n') {
+				line_count++;
+			} else if (lines[i] == '\0') {
+				break;
+			}
+			i++;
+		}
+	}
+
+	int current_line = 0;
 	nl = strchr (odata, '\n');
 	if (nl) {
 		r_cons_break_push (NULL, NULL);
 		do {
+			r_print_progressbar_with_count (core->print, current_line++, line_count, 80, true);
```

Comment:
I would add a config var to toggle this. I dont think we want progressbars all the time. Only for slow or large scripts maybe